examples/mongo_odm/demo.cpp

```diff
@@ -0,0 +1,280 @@
+#include <cereal/types/vector.hpp>
+#include <mongo_odm/odm_collection.hpp>
+
+// Makes dealing with dates easier
+typedef std::chrono::system_clock sysclock_t;
+
+class Grade {
+   public:
+    std::chrono::system_clock::time_point date;
+    std::string grade;
+    int32_t score;
+
+    // A serialization function that passes in the fields to be serialized.
+    template <class Archive>
+    void serialize(Archive& ar) {
+        ar(CEREAL_NVP(date), CEREAL_NVP(grade), CEREAL_NVP(score));
+    }
+};
+
+std::ostream& operator<<(std::ostream& os, const Grade& g) {
+    std::time_t t = std::chrono::system_clock::to_time_t(g.date);
+    os << "(" << g.grade << ", " << g.score << ", " << std::ctime(&t) << ")";
+    return os;
+}
+
+class Address {
+   public:
+    std::string building;
+    std::string street;
+    std::string zipcode;
+    double lat;
+    double lng;
+
+    // A serialization function that passes in the fields to be serialized.
+    template <class Archive>
+    void serialize(Archive& ar) {
+        ar(CEREAL_NVP(building), CEREAL_NVP(street), CEREAL_NVP(zipcode), CEREAL_NVP(lat),
+           CEREAL_NVP(lng));
+    }
+};
+
+std::ostream& operator<<(std::ostream& os, const Address& a) {
+    os << "[" << a.building << " " << a.street << ", " << a.zipcode << " "
+       << "(" << a.lat << ", " << a.lng << ")]";
+    return os;
+}
+
+class Restaurant {
+   public:
+    Address address;
+    std::string borough;
+    std::string cuisine;
+    std::vector<Grade> grades;
+    std::string name;
+    std::string restaurant_id;
+
+    // A serialization function that passes in the fields to be serialized.
+    template <class Archive>
+    void serialize(Archive& ar) {
+        ar(CEREAL_NVP(address), CEREAL_NVP(borough), CEREAL_NVP(cuisine), CEREAL_NVP(grades),
+           CEREAL_NVP(name), CEREAL_NVP(restaurant_id));
+    }
+};
+
+/**
+ * A class to store aggregate statistics. In this case, it keeps track of a borough and how many
+ * restaurants are in it.
+ */
+class BoroughStats {
+   public:
+    std::string borough;
+    int count;
+
+    // A serialization function that passes in the fields to be serialized.
+    template <class Archive>
+    void serialize(Archive& ar) {
+        ar(CEREAL_NVP(borough), CEREAL_NVP(count));
+    }
+};
+
+sysclock_t::time_point make_time_point(long seconds) {
+    std::chrono::duration<double, std::ratio<1>> since_epoch_full(seconds);
+    auto since_epoch = std::chrono::duration_cast<sysclock_t::duration>(since_epoch_full);
+    return sysclock_t::time_point(since_epoch);
+}
+
+int main(int, char**) {
+    mongocxx::instance inst{};
+    mongocxx::client conn{mongocxx::uri{}};
+    auto db = conn["test"];
+    auto restaurants_col = db["restaurants"];
+    restaurants_col.delete_many({});
+    // wrap mongocxx::collection in an odm_collection
+    mongo_odm::odm_collection<Restaurant> restaurants(restaurants_col);
+
+    // Create sample restaurant object
+    Restaurant r1;
```

Comment:
Around which parts of the code? `r1` and `r2` are used throughout, so I put them outside a scope block. 
The scope blocks are primarily for variables like `cur`, `res`, etc. that get reused multiple times.